spec/rubocop/cop/lint/duplicate_methods_spec.rb

```diff
@@ -0,0 +1,189 @@
+                    '  private def some_method',
+                    '    implement 1',
+                    '  end',
+                    '  private def some_method',
+                    '    implement 2',
+                    '  end',
+                    'end'])
+    expect(cop.offenses.size).to eq(1)
+  end
+
+  it %(registers an offense for duplicate private self methods in class) do
+    inspect_source(cop,
+                   ['class A',
+                    '  private def self.some_method',
+                    '    implement 1',
+                    '  end',
+                    '  private def self.some_method',
+                    '    implement 2',
+                    '  end',
+                    'end'])
+    expect(cop.offenses.size).to eq(1)
+  end
+
+  it %(don`t registers an offense for different private methods in class) do
+    inspect_source(cop,
+                   ['class A',
+                    '  private def some_method',
+                    '    implement 1',
+                    '  end',
+                    '  private def any_method',
+                    '    implement 2',
+                    '  end',
+                    'end'])
+    expect(cop.offenses.size).to eq(0)
+  end
+
+  it %(registers an offense for duplicate protected methods in class) do
+    inspect_source(cop,
+                   ['class A',
+                    '  protected def some_method',
+                    '    implement 1',
+                    '  end',
+                    '  protected def some_method',
+                    '    implement 2',
+                    '  end',
+                    'end'])
+    expect(cop.offenses.size).to eq(1)
+  end
+
+  it %(registers 2 offenses for pair of duplicate methods in class) do
+    inspect_source(cop,
+                   ['class A',
+                    '  def some_method',
+                    '    implement 1',
+                    '  end',
+                    '  def some_method',
+                    '    implement 2',
+                    '  end',
+                    '  def any_method',
+                    '    implement 1',
+                    '  end',
+                    '  def any_method',
+                    '    implement 2',
+                    '  end',
+                    'end'])
+    expect(cop.offenses.size).to eq(2)
+  end
+
+  it %(generate 2 offenses with specified names) do
+    inspect_source(cop,
+                   ['class A',
+                    '  def some_method',
+                    '    implement 1',
+                    '  end',
+                    '  def some_method',
+                    '    implement 2',
+                    '  end',
+                    '  def any_method',
+                    '    implement 1',
+                    '  end',
+                    '  def any_method',
+                    '    implement 2',
+                    '  end',
+                    'end'])
+    expect(cop.offenses.map(&:message)).to match_array([
```

Comment:
During spec execution, there's a help method defined that lets you call `cop.messages`, same as `cop.offenses.map(&:message)`.